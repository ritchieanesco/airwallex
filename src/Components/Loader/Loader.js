import React from "react";
import { withStyles } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";
import green from "@material-ui/core/colors/green";

const styles = theme => ({
  progress: {
    margin: theme.spacing.unit * 2,
    color: green[500],
    position: "absolute",
    top: -4,
    left: 16,
    zIndex: 1
  }
});

const Loader = props => {
  const { classes, size } = props;
  return (
    <CircularProgress size={size} className={classes.progress} thickness={7} />
  );
};

export default withStyles(styles)(Loader);
