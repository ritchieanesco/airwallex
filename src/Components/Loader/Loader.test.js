import React from "react";
import renderer from "react-test-renderer";
import Loader from "./Loader";

describe("Loader", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Loader size={28} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
