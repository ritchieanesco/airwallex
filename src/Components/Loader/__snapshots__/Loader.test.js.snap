// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly 1`] = `
<div
  className="MuiCircularProgress-root-2 MuiCircularProgress-colorPrimary-5 MuiCircularProgress-indeterminate-4 Loader-progress-1"
  role="progressbar"
  style={
    Object {
      "height": 28,
      "width": 28,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg-7"
    viewBox="0 0 50 50"
  >
    <circle
      className="MuiCircularProgress-circle-8 MuiCircularProgress-circleIndeterminate-10"
      cx={25}
      cy={25}
      fill="none"
      r={20}
      strokeWidth={7}
      style={Object {}}
    />
  </svg>
</div>
`;
